doctor/lib/ocr_utils.py

```diff
@@ -0,0 +1,345 @@
+import re
+
+import pdfplumber
+from pdfplumber.ctm import CTM
+import pytesseract
+from pytesseract import Output
+import pandas as pd
+from PIL import Image
+
+
+def deskew(obj: dict) -> bool:
+    """Remove skewed text from a page
+
+    CTM stands for current transformation matrix.
+    Pdf plumber has a method to calculate the angle of text which we use here
+
+    Traditionally this is only seen in circular stamps which confuses the
+    content, or in perpendicular text of the ninth circuit courts which also
+    confuses the text.
+    """
+    if (matrix := obj.get("matrix")) is None:
+        return True
+
+    # Remove Skew
+    my_char_ctm = CTM(*matrix)
+    if my_char_ctm.skew_x != 0:
+        return False
+    return True
+
+
+def get_page_text(page: pdfplumber.PDF.pages, strip_margin: bool) -> str:
+    """Extract page text
+
+    Using pdf plumber extract out the text of the document that is not
+    skewed (ie a stamp of approval) and extract out text removing blue text
+
+    :param page: pdf plumber page
+    :param strip_margin: a flag to crop out the margin of a document and skewed content
+    :return: text
+    """
+    if strip_margin:
+        # Crop margins and remove skewed text
+        _, _, width, height = page.bbox
+        bbox = (
+            (1 / 8.5 * width),
+            (1 / 11 * height),
+            (7.5 / 8.5 * width),
+            (10 / 11 * height),
+        )
+        doc_text = (
+            page.crop(bbox)
+            .filter(deskew)
+            .extract_text(
+                layout=True, keep_blank_chars=True, y_tolerance=5, y_density=25
+            )
+        )
+    else:
+        doc_text = page.extract_text(
+            layout=True, keep_blank_chars=True, y_tolerance=5, y_density=25
+        )
+    return doc_text
+
+
+def page_images(page: pdfplumber.pdf.Page) -> bool:
+    """Does the page have images of a certain size
+
+    Meant to exclude images that might be lines
+    """
+    for img in page.images:
+        if (
+            img.get("width") / page.width * img.get("height") / page.height
+            > 0.1
+        ) or img.get("width") * img.get("width") > 10:
```

Comment:
This is the pow(width, 2) > 10, right? Why does this work? Sorry, I'm not sure I get it.